IaC/PnP/monitoring/aks/online/aks_secure_baseline/ngsa-memory/README.md

```diff
@@ -0,0 +1,113 @@
+# Deploying NGSA Memory Manually
+
+## Prerequisites
+
+- A PnP AKS Deployment with Traefik
+
+## Install NGSA-Memory
+
+- Apply the `ngsa-memory.yaml`
+
+    ```bash
+    kubectl apply -f ./ngsa-memory
+    ```
+- Apply the new traefik configuration
+
+    ```bash
+    # REPO_ROOT=Root directory of this repository
+    kubectl apply -f ${REPO_ROOT}/enterprise_scale/construction_sets/aks/online/aks_secure_baseline/workloads/baseline/traefik.yaml
+    ```
+
+## Notes
+
+- Traefik yaml changes at [./../workloads/baseline/traefik.yaml](./../workloads/baseline/traefik.yaml)
+  - Added a new RoleBinding for the ServiceAccount `traefik-ingress-controller` in `ngsa` namespace
+  - In `traefik-ingress-config` ConfigMap
+    - Added `ngsa` namespace in `traefik.toml` under `[providers.kubernetesingress]`
+    - Added a new middleware to strip the `/memory` and `/cosmos` path prefix from the URL under `[http.middlewares.ngsa-stripprefix.stripPrefix]`
+
+
+## Deploying another app
+
+To deploy another application (say App-A) in another namespace (app-ns), these steps needs to be taken:
+
+- Traefik Configmap: `traefik-ingress-config` in [traefik.yaml](./../workloads/baseline/traefik.yaml) 
+  - Modify the `traefik-ingress-config` ConfigMap in [traefik.yaml](./../workloads/baseline/traefik.yaml) file and add the `app-ns` in `namespaces` under `[providers.kubernetesingress]`
+
+    ```toml
+    [providers.kubernetesingress]
+    ingressClass = "traefik-internal"
+    namespaces = ["a0008", "ngsa", "app-ns"]
+    ```
+
+  - Add new middleware if the app needs to be path prefixed
+    - The whole url will get passed to the app, so if the app can handle that no need to strip the path
+    - Otherwise, path needs to be stripped
+    - Add the app's path to the ConfigMap under `[http.middlewares.ngsa-stripprefix.stripPrefix]`
```

Comment:
Alternatively we could have different middlewares responsible for stripping specific prefixes. Not sure which would be easier to maintain

Reply:
I think that given the small number of prefixes in this example, using the same middleware is fine for now. If we intend to deploy multiple workloads with different prefixes, we can add new prefixes as needed.